script.js

```diff
@@ -1,406 +1,395 @@
+const allGenresBtn = document.getElementById("all-genres");
+const fictionBtn = document.getElementById("fiction");
+const sciFiBtn = document.getElementById("science-fiction");
+const fantasyBtn = document.getElementById("fantasy");
+const adventureBtn = document.getElementById("adventure");
+const horrorBtn = document.getElementById("horror");
+const mysteryBtn = document.getElementById("mystery");
+const dystopianBtn = document.getElementById("dystopian");
+const newestBtn = document.getElementById("sort-newest");
+const oldestBtn = document.getElementById("sort-oldest");
+const sortAZBtn = document.getElementById("sort-az");
+const sortZABtn = document.getElementById("sort-za");
+const filterButtons = document.querySelectorAll(".filter");
+const sortButtons = document.querySelectorAll(".sort");
+
+/////////////////////////// Global Variables ///////////////////////////
+
+let nowDispBooks = [];
+
+////////////////////////////// Functions ///////////////////////////////
+
+// Create a book card divs
+const createBookCard = (book) => {
+  let bookDiv = document.createElement("div");
+  bookDiv.setAttribute("class", "book-display");
+  bookDiv.innerHTML = `
+    <img src="${book.image}" alt="${book.title}" />
+    <h2>${book.title}</h2>
+    <hr />
+    <h5>Author: ${book.author}</h5>
+    <h5>Published: ${book.year}</h5>
+    <h5>Genre: ${book.genre}</h5>
+    <h5>Rating: ${book.rating}</h5>
+    <hr />  
+    <h4>Description</h4>
+    <p>${book.description}</p>`;
+  flexContainer.appendChild(bookDiv);
+};
+
+// Display all of the book cards
+const displayBooks = (bookArr) => {
+  bookArr.forEach((book) => {
+    createBookCard(book);
+  });
+};
+
+// Function that removes all book cards
+const removeBooks = () => {
+  flexContainer.innerHTML = "";
+};
+
+// Display all books when page is loaded
+window.onload = (event) => {
+  displayBooks(books);
+  allGenresBtn.classList.add("clicked-filter");
+  nowDispBooks = books;
+};
+
+// Function that lets us filter by genre of book
+const filterByGenre = (bookArray, genre) => {
+  return bookArray.filter((book) => book.genre === genre);
+};
+
+// Fuction that filters depending on which filter button is clicked
+const handleGenreFilterClick = (genre) => {
+  removeBooks();
+  filterButtons.forEach((btn) => btn.classList.remove("clicked-filter"));
+  // Because one of the genres is Science Fiction, we need to replace the space between the words with a -
+  const formattedGenre = genre.toLowerCase().replace(/\s+/g, "-");
+  const clickedButton = document.getElementById(`${formattedGenre}`);
+  clickedButton.classList.add("clicked-filter");
+  const filteredBooks = filterByGenre(books, genre);
+  displayBooks(filteredBooks);
+  nowDispBooks = filteredBooks;
+};
+
+// Function to get a random number
+const getRandomNum = (min, max) => {
+  // prettier-ignore
+  let getRandom = Math.floor((Math.random() * (max - min + 1) + min));
+  return getRandom;
```

Comment:
This variable name is a verb, it could maybe be mistaken for being a function. Maybe instead name it randomNumber, for example.